src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -794,18 +766,16 @@ int* mk_displ(int* cnts) {
 }
 
 static PyObject* char2pylist(char* buf, int np, int* cnt, int* displ) {
-    PyObject* plist = PyList_New(np);
-    assert(plist != NULL);
+    nb::list plist;
     for (int i = 0; i < np; ++i) {
         if (cnt[i] == 0) {
-            Py_INCREF(Py_None);  // 'Fatal Python error: deallocating None' eventually
-            PyList_SetItem(plist, i, Py_None);
+            plist[i] = nb::none();
         } else {
-            PyObject* p = unpickle(buf + displ[i], cnt[i]);
-            PyList_SetItem(plist, i, p);
+            nb::object p = unpickle(buf + displ[i], cnt[i]);
+            plist[i] = p;
         }
     }
-    return plist;
+    return plist.ptr();
```

Comment:
Why doesn't the reference count of `plist.ptr()` drop to `0` when `plist` goes out of scope?

Reply:
it does sorry ;)